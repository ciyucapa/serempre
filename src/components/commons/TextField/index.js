import React from 'react';
import PropTypes from 'prop-types';

import {sizesFonts} from '../../../utils';

export const HIGH_SIZE = 'high';
export const NORMAL_SIZE = 'normal';
export const SMALL_SIZE = 'small';

const TextField = (props) => {
    const {text, type} = props;
    return (
        <div style={{
            fontSize: sizesFonts[type],
        }}>
            {text}
        </div>
    );
};

TextField.propTypes = {
  text: PropTypes.string,
  type: PropTypes.string,
};

TextField.defaultProps = {
  text: '',
  type: NORMAL_SIZE,
};

export default TextField;
