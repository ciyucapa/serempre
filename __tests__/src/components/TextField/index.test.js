import React from 'react';
import renderer from 'react-test-renderer';

import TextField, {NORMAL_SIZE} from '../../../../src/components/commons/TextField';

describe('TextField', () => {
    it('renders correctly', () => {
        const props = {
            text: 'test',
            type: NORMAL_SIZE,
            style: {
                paddingBottom: 20,
            },
        };

        const tree = renderer
            .create(<TextField {...props} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
