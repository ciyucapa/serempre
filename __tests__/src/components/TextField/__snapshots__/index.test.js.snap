// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField renders correctly 1`] = `
<div
  style={
    Object {
      "fontSize": 12,
      "paddingBottom": 20,
    }
  }
>
  test
</div>
`;
